src/extractor/extractor_callbacks.cpp

```diff
@@ -31,11 +31,15 @@ namespace extractor
 using TurnLaneDescription = guidance::TurnLaneDescription;
 namespace TurnLaneType = guidance::TurnLaneType;
 
-ExtractorCallbacks::ExtractorCallbacks(ExtractionContainers &extraction_containers)
-    : external_memory(extraction_containers)
+ExtractorCallbacks::ExtractorCallbacks(ExtractionContainers &extraction_containers,
+                                       guidance::LaneDescriptionMap &lane_description_map)
+    : lane_description_map(lane_description_map), external_memory(extraction_containers)
 {
     // we reserved 0, 1, 2 for the empty case
     string_map[MapKey("", "")] = 0;
+
+    // The map should be empty before we start initializing it
+    BOOST_ASSERT(lane_description_map.empty());
```

Comment:
Hm doesn't it make more sense then to keep the map a member attribute and add a function so that the call site eventually can get access to it? (We destroy the callbacks, see comment above, but you can always move out the map)